packages/command-not-found/command-not-found.cpp

```diff
@@ -0,0 +1,213 @@
+#include <iostream>
+#include <cstring>
+#include <map>
+#include <list>
+
+using namespace std;
+
+list<string> games_commands = {
+#ifdef __aarch64__
+# include "games/commands-aarch64.h"
+#elif defined __arm__
+# include "games/commands-arm.h"
+#elif defined __i686__
+# include "games/commands-i686.h"
+#elif defined __x86_64__
+# include "games/commands-x86_64.h"
+#else
+# error Failed to detect arch
+#endif
+};
+
+list<string> main_commands = {
+#ifdef __aarch64__
+# include "main/commands-aarch64.h"
+#elif defined __arm__
+# include "main/commands-arm.h"
+#elif defined __i686__
+# include "main/commands-i686.h"
+#elif defined __x86_64__
+# include "main/commands-x86_64.h"
+#else
+# error Failed to detect arch
+#endif
+};
+
+list<string> root_commands = {
+#ifdef __aarch64__
+# include "root/commands-aarch64.h"
+#elif defined __arm__
+# include "root/commands-arm.h"
+#elif defined __i686__
+# include "root/commands-i686.h"
+#elif defined __x86_64__
+# include "root/commands-x86_64.h"
+#else
+# error Failed to detect arch
+#endif
+};
+
+list<string> science_commands = {
+#ifdef __aarch64__
+# include "science/commands-aarch64.h"
+#elif defined __arm__
+# include "science/commands-arm.h"
+#elif defined __i686__
+# include "science/commands-i686.h"
+#elif defined __x86_64__
+# include "science/commands-x86_64.h"
+#else
+# error Failed to detect arch
+#endif
+};
+
+list<string> unstable_commands = {
+#ifdef __aarch64__
+# include "unstable/commands-aarch64.h"
+#elif defined __arm__
+# include "unstable/commands-arm.h"
+#elif defined __i686__
+# include "unstable/commands-i686.h"
+#elif defined __x86_64__
+# include "unstable/commands-x86_64.h"
+#else
+# error Failed to detect arch
+#endif
+};
+
+list<string> x11_commands = {
+#ifdef __aarch64__
+# include "x11/commands-aarch64.h"
+#elif defined __arm__
+# include "x11/commands-arm.h"
+#elif defined __i686__
+# include "x11/commands-i686.h"
+#elif defined __x86_64__
+# include "x11/commands-x86_64.h"
+#else
+# error Failed to detect arch
+#endif
+};
+
+struct info {string owner, repository;};
+
+inline int termux_min3(int a, int b, int c) {
+	return (a < b ? (a < c ? a : c) : (b < c ? b : c));
+}
+
+int termux_levenshtein_distance(char const* s1, char const* s2) {
+  int s1len = strlen(s1);
+  int s2len = strlen(s2);
+  int x, y;
+  int **matrix;
+  int distance;
+  matrix = (int **) malloc(sizeof *matrix * (s2len+1));
+  if (matrix) {
+    matrix[0] = (int *) malloc(sizeof *matrix[0] * (s1len+1));
+  } else {
+    cerr << "Memory allocation seem to have failed" << endl;
+    return -2;
+  }
+  if (matrix[0]) {
+    matrix[0][0] = 0;
+  } else {
+    cerr << "Memory allocation seem to have failed" << endl;
+    return -3;
+  }
+  for (x = 1; x <= s2len; x++) {
+    matrix[x] = (int *) malloc(sizeof *matrix[x] * (s1len+1));
+    matrix[x][0] = matrix[x-1][0] + 1;
```

Comment:
> if the malloc done in termux_levenshtein_distance fails then the program will probably segfault in main() after returning.

It will segfault here.

Reply:
Added check here.